modules/aioseop_sitemap.php

```diff
@@ -3376,91 +3396,312 @@ private function get_images_from_post( $post ) {
+					$rtn_images[] = array(
+						'image:loc'     => $this->clean_url( $v1_image['url'] ),
+						'image:caption' => wp_get_attachment_caption( $v1_image['id'] ),
+						'image:title'   => get_the_title( $v1_image['id'] ),
 					);
 				}
+			} else {
+				// Reset if new post is found.
+				$query_step = 0;
+				if ( is_multisite() ) {
+					set_site_transient( 'multisite_aioseop_posts_images_query_step_' . $current_sitemap, ( $query_step + 1 ), 24 * HOUR_IN_SECONDS );
+				} else {
+					set_transient( 'aioseop_posts_images_query_step_' . $current_sitemap, ( $query_step + 1 ), 24 * HOUR_IN_SECONDS );
+				}
 			}
 
-			return $images;
+			return $rtn_images;
+		}
+
+		/**
+		 * Set Transient Post Parent Images
+		 *
+		 * @since 2.9.2
+		 */
+		public function set_transient_post_parents_images() {
+			$current_sitemap = sanitize_key( $_SERVER['REQUEST_URI'] );
+			if ( is_multisite() ) {
+				set_site_transient( 'multisite_aioseop_posts_images_' . $current_sitemap, $this->post_parents_images, 24 * HOUR_IN_SECONDS );
+			} else {
+				set_transient( 'aioseop_posts_images_' . $current_sitemap, $this->post_parents_images, 24 * HOUR_IN_SECONDS );
+			}
 		}
 
 		/**
-		 * Fetch image attributes such as title and caption given the image URL.
+		 * Query Post Images Step
+		 *
+		 * This performs 3 different queries depending on the current series of instances it may be on since it was
+		 * initially set with transient data; data expires in 24 hours.
 		 *
-		 * @param string $url The image URL.
+		 * 1. Uses optimal and essential functions.
+		 * 2. Fallback with simular functionality which step 1 may have missed.
+		 * 3. Heavy functionality for deeper scans; most of which converts url to id and should be avoided if possible.
+		 *
+		 * @param int     $query_step
+		 * @param WP_Post $post
+		 * @param array   $current_images
 		 * @return array
 		 */
-		private function get_image_attributes( $url ) {
-			global $wpdb;
-			$attributes = array();
-
-			$attachment = $wpdb->get_col( $wpdb->prepare( "SELECT ID FROM $wpdb->posts WHERE guid='%s';", $url ) );
-			if ( $attachment && is_array( $attachment ) && is_numeric( $attachment[0] ) ) {
-				$attributes = array(
-					'image:caption' => wp_get_attachment_caption( $attachment[0] ),
-					'image:title'   => get_the_title( $attachment[0] ),
-				);
+		public function query_post_images_step( $query_step, $post, $current_images = array() ) {
+			$rtn_images = array();
+			$query_step = intval( $query_step );
+
+			switch ( $query_step ) {
+				case 0:
+					// GET BUILT-IN ATTACHMENT OBJECTS.
+					// Query_Step 0 may include Post Thumbnails, but there are a few objects that could be miss.
+					$args = array(
+						'post_type'   => 'attachment',
+						'numberposts' => 1000, //-1, // It's unlikely there will be over 1000 image on a single page.
+						'post_status' => null,
+						'post_parent' => $post->ID,
+					);
+
+					$attachments = get_posts( $args );
```

Comment:
use WP_Query here as get_posts is inefficient. set cache params to false as well